hw/ip/aes/dv/env/aes_scoreboard.sv

```
@@ -88,16 +88,17 @@ class aes_scoreboard extends cip_base_scoreboard #(
       csr_name = csr.get_name();
       case (1)
         // add individual case item for each csr
-         (!uvm_re_match("ctrl", csr_name)): begin
-          input_item.manual_op = item.a_data[7];
-          input_item.key_len   = item.a_data[6:4];
+         (!uvm_re_match("ctrl_shadowed", csr_name)): begin
+          input_item.manual_op = item.a_data[8];
+          input_item.key_len   = item.a_data[7:5];
           `downcast(input_item.operation, item.a_data[0]);
-          case (item.a_data[3:1])
-            3'b001:  input_item.mode = AES_ECB;
-            3'b010:  input_item.mode = AES_CBC;
-            3'b100:  input_item.mode = AES_CTR;
-            default: input_item.mode = AES_ECB;
-          endcase // case item.a_data[3
+          case (item.a_data[4:1])
+            4'b0001:  input_item.mode = AES_ECB;
+            4'b0010:  input_item.mode = AES_CBC;
+            4'b0100:  input_item.mode = AES_CTR;
+            4'b1000:  input_item.mode = AES_NONE;
+            default:  input_item.mode = AES_ECB;
```

Should we make this none to match the RTL?
I can do that.

---

If this is an invalid mode, is that actually allowed by the module spec? If not, surely the test code should just throw an error?

---

yes this mode is allowed by the RTL
so the dv should also go into this state.
the question here is  since the RTL goes into this as a default state - the DV should also.

---

Yep, AES_NONE is the reset value in the RTL and every invalid setting is mapped to AES_NONE. I wasn't fully sure how to correctly adapt DV for this. If you could change that @rasmus-madsen this would be great. This should possibly be a follow-up PR?

---

yep let's do that in a follow up PR